process/block/interceptedBlocks/interceptedBlockHeader.go

```diff
@@ -0,0 +1,159 @@
+package interceptedBlocks
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+// InterceptedHeader represents the wrapper over HeaderWrapper struct.
+// It implements Newer and Hashed interfaces
+type InterceptedHeader struct {
+	hdr                 *block.Header
+	marshalizer         marshal.Marshalizer
+	hasher              hashing.Hasher
+	multiSigVerifier    crypto.MultiSigVerifier
+	chronologyValidator process.ChronologyValidator
+	shardCoordinator    sharding.Coordinator
+	hash                []byte
+	isForMyShard        bool
+}
+
+// NewInterceptedHeader creates a new instance of InterceptedHeader struct
+func NewInterceptedHeader(arg *ArgInterceptedBlockHeader) (*InterceptedHeader, error) {
+	if arg == nil {
+		return nil, process.ErrNilArguments
+	}
+	if arg.HdrBuff == nil {
+		return nil, process.ErrNilBuffer
+	}
+	if check.IfNil(arg.Marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+	if check.IfNil(arg.Hasher) {
+		return nil, process.ErrNilHasher
+	}
+	if check.IfNil(arg.MultiSigVerifier) {
+		return nil, process.ErrNilMultiSigVerifier
+	}
+	if check.IfNil(arg.ChronologyValidator) {
+		return nil, process.ErrNilChronologyValidator
+	}
+	if check.IfNil(arg.ShardCoordinator) {
+		return nil, process.ErrNilShardCoordinator
+	}
+
+	hdr := &block.Header{
+		MiniBlockHeaders: make([]block.MiniBlockHeader, 0),
+		MetaBlockHashes:  make([][]byte, 0),
+	}
+	err := arg.Marshalizer.Unmarshal(hdr, arg.HdrBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	inHdr := &InterceptedHeader{
+		hdr:                 hdr,
+		marshalizer:         arg.Marshalizer,
+		hasher:              arg.Hasher,
+		multiSigVerifier:    arg.MultiSigVerifier,
+		chronologyValidator: arg.ChronologyValidator,
+		shardCoordinator:    arg.ShardCoordinator,
+	}
+	inHdr.processFields(arg.HdrBuff)
+
+	return inHdr, nil
+}
+
+func (inHdr *InterceptedHeader) processFields(txBuff []byte) {
+	inHdr.hash = inHdr.hasher.Compute(string(txBuff))
+
+	isHeaderForCurrentShard := inHdr.shardCoordinator.SelfId() == inHdr.HeaderHandler().GetShardID()
+	isMetachainShardCoordinator := inHdr.shardCoordinator.SelfId() == sharding.MetachainShardId
```

Comment:
this is constant for a complete epoch. GetShardID could return metachainShardID and would be ok.

Reply:
So what should I do?

Reply:
can be set on constructor.